go/login/login_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package login
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/google/glome/go/glome"
+)
+
+const (
+	tvsCnt = 2
+)
+
+var serviceKeyIds = []uint8{1, 0}
+
+func handleError(e error, t *testing.T) {
+	if e != nil {
+		t.Fatalf("Unexpected Error: " + e.Error())
+	}
+}
+
+type testVector struct {
+	kap        []byte
+	ka         []byte
+	kbp        []byte
+	kb         []byte
+	ks         []byte
+	prefix     byte
+	hostIdType string
+	hostId     string
+	action     string
+	msg        []byte
+	url        string
+	prefixN    []byte
+	tag        []byte
+	token      string
+}
+
+type KeyPair struct {
+	priv glome.PrivateKey
+	pub  glome.PublicKey
+}
+
+func decodeString(s string) []byte {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		panic(fmt.Sprintf("Invalid hexadecimal string %v input in test", s))
+	}
+	return b
+}
+
+func keys(t *testing.T, kp []byte, k []byte) *KeyPair {
+	aPriv, err := glome.PrivateKeyFromSlice(kp)
+	handleError(err, t)
+	aPub, err := glome.PublicKeyFromSlice(k)
+	handleError(err, t)
+	return &KeyPair{*aPriv, *aPub}
+}
+
+func (tv *testVector) Dialog(t *testing.T) (*glome.Dialog, *glome.Dialog) {
+	clientKP := keys(t, tv.kap, tv.ka)
+	serverKP := keys(t, tv.kbp, tv.kb)
+
+	sending, err := clientKP.priv.Exchange(&serverKP.pub)
+	handleError(err, t)
+	receiving, err := serverKP.priv.Exchange(&clientKP.pub)
+	handleError(err, t)
+
+	return sending, receiving
+}
+
+func testVectors() []testVector {
+	prefix1, _ := strconv.Atoi("1")
+	prefix2, _ := strconv.Atoi("0x51")
```

Comment:
Never swallow errors. In this case I'm fairly sure 0x51 is not a legit number? godoc says "Atoi is equivalent to ParseInt(s, 10, 0), converted to type int." and you are looking for base 16. And indeed https://play.golang.org/p/hOHLM8F9rVJ throws an error.

Reply:
Fixed for testing. Need to have a look at login.go